src/code/z_kaleido_manager.c

```diff
@@ -1,15 +1,103 @@
 #include "global.h"
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_kaleido_manager/D_801DF9C0.s")
+#define KALEIDO_OVERLAY(name)                                                                                \
+    {                                                                                                        \
+        NULL, (u32)_ovl_##name##SegmentRomStart, (u32)_ovl_##name##SegmentRomEnd, _ovl_##name##SegmentStart, \
+            _ovl_##name##SegmentEnd, 0, #name,                                                               \
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_kaleido_manager/func_80163700.s")
+KaleidoMgrOverlay gKaleidoMgrOverlayTable[] = {
+    KALEIDO_OVERLAY(kaleido_scope),
+    KALEIDO_OVERLAY(player_actor),
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_kaleido_manager/func_80163758.s")
+void* sKaleidoAreaPtr = NULL;
+KaleidoMgrOverlay* gKaleidoMgrCurOvl = NULL;
+FaultAddrConvClient sKaleidoAreaFaultClient;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_kaleido_manager/func_801637B4.s")
+void* func_80163700(void* arg0, void* arg1) {
```

Comment:
Graph called there similar function `Graph_FaultAddrConvFunc` so perhaps this could be `KaleidoManager_FaultAddrConvFunc`. They also have `arg0` as `address` and `arg1` as`param`.